### ISSUE TITLE
Ability to select PMD version

### ISSUE THREAD

**Issue by sebastianhaberey** [amitdev/PMD-Intellij#63]
Hi! First off, thank you vor PMD-IntelliJ, we are using it in production here and it works fine! If I had one wish, it would be to be able to choose the PMD version that PMD-Intellij uses for checking, like the CheckStyle-IDEA plugin does:

![checkstyle-intellij](https://user-images.githubusercontent.com/4152849/46852779-5f5a3480-cdfc-11e8-9383-5fc29b8f4a2b.PNG)

The problem is that we like to keep our plugins up-to-date and when a new PMD-IntelliJ-Version comes out, everybody clicks "Update". But then, newly introduced PMD warnings pop up and we have a "no-warnings" check-in policy.

So now, someone has to review the newly introduced PMD rules and configure them according to team preferences. Also, the maven PMD configuration needs to be updated so the warnings on the build server match the warnings in IntelliJ. Sure, we're going to do this eventually, but we'd like to be able to do this as a planned item and not in the middle of a sprint.

There's an option in IntelliJ to configure the required plugin version for a project, but it doesn't seem to work as expected. I filed [an issue for that](https://youtrack.jetbrains.com/issue/IDEA-199249), but I don't think it has top priority.

**tl;dr** updating PMD-IntelliJ often leads to new PMD warnings, creating unplanned work, user-stipulated PMD version would help

**Comment by rbieniek** [comment]
Another point here supporting the request is that the currently used PMD version is not handling recently introduced rules which came along with a newer version of PMD, for example 'AvoidMessageDigestField'

**Comment by triestelaporte** [comment]
I'd settle for a plugin that used PMD 6.22+ 😁  But specifying version would be nice.

**Comment by amitdev** [comment]
Updated to 6.25. Adding support to choose PMD version is tricky and I don't have time to look at it now. Happy to accept PRs though :)

**Comment by jborgers** [comment]
you can always download the desired release from GitHub and install the plugin in IntelliJ from file. Would that help?

**Comment by sebastianhaberey** [comment]
Wow this is ancient 😄 IIRC that doesn't help since IntelliJ would still offer to update the manually installed plugin every time a new PMD version comes out.

**Comment by sebastianhaberey** [comment]
> So actually, you want to be able to define the set of pmd standard rules (and custom rules) for a project, separately from the version of PMD, the version of the PMD-Plugin and the version of IntelliJ, right?

Yes, that's it. "Project X has rule set Y" and that would stay untouched during updates etc.

**Comment by adangel** [comment]
One intermediate/first step solution could be, to have a configuration for a "PMD Path" available: One could then manually download a full PMD distribution, extract it and point the IntelliJ Plugin to this path. It would then load PMD from this path instead from the jars shipped within the plugin. The "PMD Path" config property could also be stored in the "PMDPlugin.xml" file in the project, thus it could be shared (provided, that the PMD Path is the same for all team members - so not possible for cross-os team members).

The more sophisticated solution would be similar what checkstyle does: actually provide multiple versions of PMD inside the plugin. This is more complicated, if we want to make efficient use of disk space - thus only added dependencies once and reuse these across different PMD versions where possible to minimize the plugin size.
